mathesar_ui/src/pages/record/Widgets.svelte

```diff
@@ -0,0 +1,46 @@
+<script lang="ts">
+  import type { JoinableTablesResult } from '@mathesar/api/tables/joinable_tables';
+  import TableWidget from './TableWidget.svelte';
+
+  export let recordId: number;
+  export let joinableTablesResult: JoinableTablesResult;
+
+  $: tableNameMap = new Map(
+    Object.entries(joinableTablesResult.tables).map(([tableId, table]) => [
+      parseInt(tableId, 10),
+      table.name,
+    ]),
+  );
+  $: columnNameMap = new Map(
+    Object.entries(joinableTablesResult.columns).map(([columnId, column]) => [
+      parseInt(columnId, 10),
+      column.name,
+    ]),
+  );
+  $: tableWidgetInputs = joinableTablesResult.joinable_tables
+    .filter((joinableTable) => joinableTable.multiple_results)
+    .map((joinableTable) => ({
+      table: {
+        id: joinableTable.target,
+        name: tableNameMap.get(joinableTable.target) ?? '(unknown table)',
+      },
+      fkColumn: {
+        id: joinableTable.jp_path[0].slice(-1)[0],
+        name:
+          columnNameMap.get(joinableTable.jp_path[0].slice(-1)[0]) ??
+          '(unknown column)',
+      },
+    }));
+</script>
+
+{#each tableWidgetInputs as { table, fkColumn }}
```

Comment:
A key is required here for better performance.

Reply:
Thanks. Fixed in 9a54f6740d59408deab3deceea9658fec4ea7064.

FWIW, I went looking for a way to lint away this problem and found [this sad thread rejecting the idea](https://github.com/sveltejs/eslint-plugin-svelte3/issues/42). Just like the OP on that thread, I'm accustomed to the Vue linter that complains about missing keys.